File: Client/CredentialService.cs
using KolibSoft.AuthStore.Core.Filters;
using KolibSoft.AuthStore.Core.Models;
using KolibSoft.Catalogue.Client;

namespace KolibSoft.AuthStore.Client;

public class CredentialService : CatalogueService<CredentialModel, CredentialFilters>
{
    public CredentialService(HttpClient httpClient, string uri) : base(httpClient, uri) { }
}

File: docs/lib/services/credential_service.js
import { CredentialModel } from "../models/credential_model.js";
import { CatalogueService } from "../modules/catalogue.js"

class CredentialService extends CatalogueService {

    /**
     * @param {(json: {}) => CredentialModel} creator 
     * @param {(input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response>} fetch 
     * @param {string} uri 
     */
    constructor(creator, fetch, uri) {
        super(creator, fetch, uri);
    }

}

export {
    CredentialService
}